scripts/irods/test/test_irepl.py

```diff
@@ -11,9 +12,140 @@
+        super(Test_Irepl, self).tearDown()
+
+    def do_test_for_issue_5592(self, trigger_checksum_mismatch_error):
+        def get_physical_path(data_object, replica_number):
+            gql = "select DATA_PATH where COLL_NAME = '{0}' and DATA_NAME = '{1}' and DATA_REPL_NUM = '{2}'"
+            path, err, ec = self.admin.run_icommand(['iquest', '%s', gql.format(self.admin.session_collection, data_object, replica_number)])
+            self.assertEqual(ec, 0)
+            self.assertEqual(len(err), 0)
+            self.assertGreater(len(path.strip()), 0)
+            return path.strip()
+
+        def get_replica_status(data_object, replica_number):
+            gql = "select DATA_REPL_STATUS where COLL_NAME = '{0}' and DATA_NAME = '{1}' and DATA_REPL_NUM = '{2}'"
+            status, err, ec = self.admin.run_icommand(['iquest', '%s', gql.format(self.admin.session_collection, data_object, replica_number)])
+            self.assertEqual(ec, 0)
+            self.assertEqual(len(err), 0)
+            self.assertGreater(len(status.strip()), 0)
+            return status.strip()
+
+        def get_checksum(data_object, replica_number, expected_length_of_checksum=49):
+            gql = "select DATA_CHECKSUM where COLL_NAME = '{0}' and DATA_NAME = '{1}' and DATA_REPL_NUM = '{2}'"
+            checksum, err, ec = self.admin.run_icommand(['iquest', '%s', gql.format(self.admin.session_collection, data_object, replica_number)])
+            self.assertEqual(ec, 0)
+            self.assertEqual(len(err), 0)
+            self.assertEqual(len(checksum.strip()), expected_length_of_checksum)
+            return checksum.strip()
+
+        # Define our resource names for the "finally" block.
+        repl_resc = 'issue_5592_repl_resc'
+        ufs_resources = ['issue_5592_ufs1', 'issue_5592_ufs2', 'issue_5592_ufs3']
+
+        # Define the name of the data object for the "finally" block.
+        data_object = 'foo'
+
+        try:
+            # Create our resource hierarchy.
+            lib.create_replication_resource(repl_resc, self.admin)
+
+            for resc_name in ufs_resources:
+                lib.create_ufs_resource(resc_name, self.admin)
+                self.admin.assert_icommand(['iadmin', 'addchildtoresc', repl_resc, resc_name])
+
+            # Create a data object (this should result in three replicas).
+            self.admin.assert_icommand(['istream', 'write', '-R', repl_resc, data_object], input='bar')
+            self.admin.assert_icommand(['ils', '-l', data_object], 'STDOUT', [
+                ' {};{}            3 '.format(repl_resc, ufs_resources[0]),
+                ' {};{}            3 '.format(repl_resc, ufs_resources[1]),
+                ' {};{}            3 '.format(repl_resc, ufs_resources[2])
+            ])
+
+            # Checksum the first replica.
+            # TODO Should ALL of the replicas be checksum'd?
+            self.admin.assert_icommand(['ichksum', '-n', '0', data_object], 'STDOUT', ['sha2:'])
```

Comment:
@trel Any thoughts on calculating checksums for one or all replicas?

Also, I'm going to remove the `-n0` on the `ichksum` call because that will bring the test closer to what was described in the issue. The downside is that we are relying on the server to vote and pick the replica with the checksum as the replication source. What if the voting picks a replica without a checksum.

Any thoughts on that?

Reply:
We want to test what happens when there is a checksum mismatch on the replication... so i think the `-a` makes sense here.   I don't know that testing the case of replication without a checksum tells us anything.

Reply:
Cool. Will add the -a